Clarify SQS visibility timeout and duplication protection in S3 source (#12915)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sources/s3.md

:--- | :--- | :--- | :---
`queue_url` | Yes | String | The URL of the Amazon SQS queue from which messages are received.
`maximum_messages` | No | Integer | The maximum number of messages to receive from the Amazon SQS queue in any single request. Default is `10`.

> `visibility_timeout` | No | Duration | The visibility timeout to apply to messages read from the Amazon SQS queue. This should be set to the amount of time that Data Prepper may take to read all the S3 objects in a batch. If processing exceeds this timeout and `visibility_duplication_protection` is not enabled, the message becomes visible in the queue again and may be reprocessed, which can create duplicate data in the sink. The maximum value is 12 hours (`43200s`). Default is `30s`.

Reviewer: ```suggestion
`visibility_timeout` | No | Duration | The visibility timeout to apply to messages read from the Amazon SQS queue. This parameter should be set to the amount of time that Data Prepper may take to read all the S3 objects in a batch. If processing exceeds this timeout and `visibility_duplication_protection` is not enabled, the message becomes visible in the queue again and may be reprocessed, which can create duplicate data in the sink. The maximum value is 12 hours (`43200s`). Default is `30s`.
```

`wait_time` | No | Duration | The amount of time to wait for long polling on the Amazon SQS API. Default is `20s`.
`poll_delay` | No | Duration | A delay placed between the reading and processing of a batch of Amazon SQS messages and making a subsequent request. Default is `0s`.

> `visibility_duplication_protection` | No | Boolean | If set to `true`, Data Prepper attempts to avoid duplicate processing by extending the visibility timeout of SQS messages. Until the data reaches the sink, Data Prepper regularly calls `ChangeMessageVisibility` to avoid rereading of the S3 object. This option takes effect only when `acknowledgments` is enabled on the `s3` source, because the visibility timeout is extended through end-to-end acknowledgments. When `acknowledgments` is not enabled, the visibility timeout is not extended even if this option is set to `true`. To use this feature, you need to grant permissions to `sqs:ChangeMessageVisibility` on the IAM role. Default is `false`.

Reviewer: ```suggestion
`visibility_duplication_protection` | No | Boolean | If set to `true`, Data Prepper attempts to avoid duplicate processing by extending the visibility timeout of SQS messages. Until the data reaches the sink, Data Prepper regularly calls `ChangeMessageVisibility` to avoid rereading the S3 object. This option takes effect only when `acknowledgments` is enabled on the `s3` source, because the visibility timeout is extended through end-to-end acknowledgments. When `acknowledgments` is not enabled, the visibility timeout is not extended even if this option is set to `true`. To use this feature, you must grant permissions to `sqs:ChangeMessageVisibility` on the IAM role. Default is `false`.
```

> `visibility_duplicate_protection_timeout` | No | Duration | Sets the maximum total length of time that a message will not be processed when using `visibility_duplication_protection`. The minimum value is `30s` and the maximum value is 24 hours. Defaults to two hours.

Reviewer: ```suggestion
`visibility_duplicate_protection_timeout` | No | Duration | Sets the maximum total length of time that a message will not be processed when using `visibility_duplication_protection`. The minimum value is `30s` and the maximum value is 24 hours (`86400s`). Default is 2 hours (`7200s`).
```
